app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,52 @@
+class RestaurantsController < ApplicationController
+  before_action :set_restaurant, only: [:show, :update, :destroy]
+
+  # GET /restaurants
+  def index
+    @restaurants = Restaurant.all
+
+    render json: @restaurants
+  end
+
+  # GET /restaurants/1
+  def show
+    @restaurants = Restaurant.find(params[:id])
+    render json: @restaurant
+  end
+
+  # POST /restaurants
+  def create
+    @restaurant = Restaurant.new(restaurant_params)
+
+    if @restaurant.save
```

Comment:
This should probably be outside:
```
did_creation_succeed = @restaurant.save
if did_reation_succeed
   ....
```
Note that save only returns `false` on validation errors (when `valid?` returns `false`). If other errors occur at the database level, like a database deadlock or trying to insert null into a column that doesn’t allow it, that will still raise an exception - in that case it means you'll never get to your `else` clause

Another suggestion is to use `create` instead of this entire flow:
https://guides.rubyonrails.org/active_record_basics.html#create

Reply:
done.